pkg/reconciler/inmemorychannel/dispatcher/controller.go

```diff
@@ -61,6 +61,11 @@ type envConfig struct {
 	// TODO: change this environment variable to something like "PodGroupName".
 	PodName       string `envconfig:"POD_NAME" required:"true"`
 	ContainerName string `envconfig:"CONTAINER_NAME" required:"true"`
+
+	// HTTP client conf used when dispatching events
+	MaxIdleConns int `envconfig:"MAX_IDLE_CONNS" required:"true"`
+	// MaxIdleConnsPerHost refers to the max idle connections per host, as in net/http/transport.
+	MaxIdleConnsPerHost int `envconfig:"MAX_IDLE_CONNS_PER_HOST" required:"true"`
```

Comment:
Optional: should we set a default value here?  The reason I am asking is that I likely don't care about these two parameters until I see a problem. And IIUC, the http client is reused when the configuration is the same, so it is even possible to have a single http client, right?

Or can you give a guideline of how to set it?

Reply:
The default value is already set both in the dispatcher yaml and coded in the config map watcher in the imc controller, i would rather avoid to set one here. If there isn't any value, we just make the dispatcher crash, which i think it's reasonable.

> And IIUC, the http client is reused when the configuration is the same, so it is even possible to have a single http client, right?

We use just a single http client across the whole dispatcher. This is a change we had to do to share a single http underlying connection pool (look https://github.com/knative/eventing/issues/4461). Note that with this PR, the actual code of kncloudevents to reuse the same client will have some unused branches (e.g. to recreate a new client after the connection args change). In a followup i'm going to clean up that code.